### PR TITLE
Harvest: Make the TriggerManager working with konnectorJob

### PR DESCRIPTION
**Pull request by CPatchane**
Since the TriggerLauncher we added a new konnectorJob object to handle all the process related to the trigger execution.
Now the konnectorJob can also handle account and trigger creation and the TriggerManager can now use it to submit the account form and run the trigger just after.

⚠️ Need to update and add more tests before merging it

### CODE REVIEW

File: packages/cozy-harvest-lib/src/components/HOCs/withKonnectorJob.jsx [hunk 1]
```diff
@@ -6,16 +6,28 @@ import KonnectorJob from '../../models/KonnectorJob'
...
 
 export const withKonnectorJob = WrappedComponent => {
   class ComponentWithKonnectorJob extends PureComponent {
     constructor(props, context) {
       super(props, context)
-      this.konnectorJob = new KonnectorJob(context.client, props.trigger)
+      this.konnectorJob = new KonnectorJob(
+        context.client,
+        props.trigger,
+        props.account
```
Comment (gregorylegarec) [inline packages/cozy-harvest-lib/src/components/HOCs/withKonnectorJob.jsx]: If we have the `trigger`, we can also have the `account`, what am I missing ?

File: packages/cozy-harvest-lib/src/components/TriggerManager.jsx [hunk 2]
```diff
@@ -31,172 +25,51 @@ const MODAL_PLACE_ID = 'coz-harvest-modal-place'
...
-      error: null,
-      status: RUNNING_TWOFA
-    })
-    try {
-      await saveAccount(konnector, accounts.updateTwoFaCode(account, code))
-      if (this.jobWatcher) this.jobWatcher.enableSuccessTimer(10000)
-    } catch (error) {
-      return this.handleError(error)
+    // prepare the connection, create account and trigger if needed
+    await konnectorJob.prepareConnection(konnector, data, t('default.baseDir'))
```
Comment (gregorylegarec) [inline packages/cozy-harvest-lib/src/components/TriggerManager.jsx]: Totally OK to externalizing business logic from TriggerManager, but it's not the KonnectorJob responsability to initialize every object. KonnectorJob should just laucnh a job for a given trigger, nothing more.

Trigger and account creation should be externalized in another business object.

File: packages/cozy-harvest-lib/src/components/HOCs/withKonnectorJob.jsx (hunk 1 above)
Comment (CPatchane) [reply packages/cozy-harvest-lib/src/components/HOCs/withKonnectorJob.jsx]: I still have to change that 👍

File: packages/cozy-harvest-lib/src/components/TriggerManager.jsx (hunk 2 above)
Comment (CPatchane) [reply packages/cozy-harvest-lib/src/components/TriggerManager.jsx]: It's now moved to utility classes 👍

**Comment by CPatchane** [comment]
[🚨 (𝐌) 🍊 [HAR] Intégration de la modale de modification dans harvest](https://trello.com/c/sMZCFERD/1327-%F0%9F%9A%A8-%F0%9D%90%8C-%F0%9F%8D%8A-har-int%C3%A9gration-de-la-modale-de-modification-dans-harvest)

**Comment by CPatchane** [comment]
I guess @gregorylegarec we can close this PR?

**Comment by Crash--** [comment]
What it the status of this PR ? @CPatchane @gregorylegarec ?

**Comment by nono** [comment]
@gregorylegarec are you OK to close this PR?

**Comment by nono** [comment]
Ping @gregorylegarec

**Comment by gregorylegarec** [comment]
Yes I am !

**Pull request closed without merging**